src/main/java/org/assertj/core/internal/Files.java

```diff
@@ -88,10 +89,15 @@ public static Files instance() {
   public void assertSameContentAs(AssertionInfo info, File actual, File expected) {
     verifyIsFile(expected);
     assertIsFile(info, actual);
+    BinaryDiffResult binaryDiffResult = null;
     try {
+      binaryDiffResult = binaryDiff.diff(actual, java.nio.file.Files.readAllBytes(expected.toPath()));
```

Comment:
In our case, I'm not sure it is possible to have different binary content but equal text content as we end comparing text content with the default charset - which makes me think that users should probably be able to set the Charset for comparing text files (not possible for the time being), that would mean adding this to `AbstractFileAssert`:
```java
public S hasSameContentAs(File expected, Charset charset)
```